skops/io/_persist.py

```diff
@@ -39,7 +39,7 @@ def _save(obj: Any) -> io.BytesIO:
     return buffer
 
 
-def dump(obj: Any, file: str) -> None:
+def dump(obj: Any, file: str | io.TextIOWrapper) -> None:
```

Comment:
is it a textiowrapper or anything which has `write`?

Reply:
It's the latter, if you prefer I can create a protocol for objects with write attributes, just thought `TextIOWrapper` is the most common case by far.

Reply:
Or not have it typed? 👻 

We also accept a `Path`.

What we really accept here, is a file-like object, or a path like object. And that's best explained in the docstring w/o any types.

![image](https://user-images.githubusercontent.com/1663864/205307165-6336e9f6-44dd-4a11-96d3-2829b19d84fa.png)

Reply:
I dug a bit further, it seems that `typing.BinaryIO` would be the correct type:

https://docs.python.org/3/library/typing.html#typing.BinaryIO

This allows mypy to flag code like the one below which uses the wrong mode:

```python
with open(file, 'w') as f:  # <= should be 'wb' mode
    dump(f)
```

WDYT?

Regarding `Path`: Yeah, that's supported, I originally didn't add it because it's not related to this PR, but I can still add it here.

Reply:
Ok, I could live with `BinaryIO` then.